server/plugin/api_test.go

```diff
@@ -249,74 +265,232 @@ func TestPlugin_handleVirtualAgentWebhook(t *testing.T) {
+							"enabled":true 
+						  }, 
+						  { 
+							 "label":"Live Agent Support.", 
+							 "value":"ce2ee85053130010cf8cddeeff7b12bf", 
+							 "enabled":true 
+						  } 
+						] 
+					  }
+					],
+					"score": 1
+				  }`,
+			},
+			expectedResponse: testutils.ExpectedResponse{
+				StatusCode: http.StatusOK,
+			},
+		},
+		"OutputText response is received from Virtual Agent": {
+			httpTest: httpTestString,
+			request: testutils.Request{
+				Method: http.MethodPost,
+				URL:    fmt.Sprintf("/api/v1%s?secret=mockWebhookSecret", PathVirtualAgentWebhook),
+				Body: `{
+					"requestId": "9ff925c1-893f-4a46-baa5-a64f4d8c89b1",
+					"message": {
+					  "text": "",
+					  "typed": true
+					},
+					"userId": "6816f79cc0a8016401c5a33be04be441",
+					"body": [
+					  {
+						"uiType": "OutputText",
+						"group": "DefaultText",
+						"value": "Successful",
+						"maskType": "NONE"
+					  }
+					],
+					"score": 1
+				  }`,
+			},
+			expectedResponse: testutils.ExpectedResponse{
+				StatusCode: http.StatusOK,
+			},
+		},
+		"Picker response is received from Virtual Agent": {
+			httpTest: httpTestString,
+			request: testutils.Request{
+				Method: http.MethodPost,
+				URL:    fmt.Sprintf("/api/v1%s?secret=mockWebhookSecret", PathVirtualAgentWebhook),
+				Body: `{
+					"requestId": "9ff925c1-893f-4a46-baa5-a64f4d8c89b1",
+					"message": {
+					  "text": "",
+					  "typed": true
+					},
+					"userId": "6816f79cc0a8016401c5a33be04be441",
+					"body": [
 						{
-							Value: OutputLink{
-								UIType: "mockUIType",
-								Group:  "mockGroup",
-								Label:  "mockLabel",
-								Header: "mockHeader",
-								Type:   "mockType",
-								Value: OutputLinkValue{
-									Action: "mockAction",
-								},
-							},
-						},
+							"uiType":"Picker",
+							"group":"DefaultPicker",
+							"required":true,
+							"nluTextEnabled":false,
+							"label":"I want to be sure I got this right. What item best describes what you want to do?",
+							"itemType":"List",
+							"style":"list",
+							"multiSelect":false,
+							"options":[
+							  {
+								"label":"Live Agent Support.",
+								"value":"Live Agent Support.",
+								"renderStyle":"data",
+								"enabled":false
+							  },
+							  {
+								"label":"Virtual Agent Capabilities.",
+								"value":"Virtual Agent Capabilities.",
+								"renderStyle":"data",
+								"enabled":false
+							  },
+							  {
+								"label":"I want something else",
+								"value":"-1",
+								"renderStyle":"data",
+								"enabled":false
+							  }
+							],
+							"scriptedData":null
+						  }
+					],
+					"score": 1
+				  }`,
+			},
+			expectedResponse: testutils.ExpectedResponse{
+				StatusCode: http.StatusOK,
+			},
+		},
+		"GroupedPartsOutputControl response is received from Virtual Agent": {
+			httpTest: httpTestString,
+			request: testutils.Request{
+				Method: http.MethodPost,
```

Comment:
Here i think method is http post for every tc so you can just write those once before the request part rather than defining this in every tc

Reply:
We are sending this whole struct to a function for creating the HTTP request. So, its needed to be present inside